extension/src/background/messageListener/popupMessageListener.ts

```diff
@@ -254,17 +254,29 @@ export const popupMessageListener = (request: Request, sessionStore: Store) => {
     password,
     keyPair,
     imported = false,
+    isSettingHashKey = false,
   }: {
     mnemonicPhrase: string;
     password: string;
     keyPair: KeyPair;
     imported?: boolean;
+    isSettingHashKey?: boolean;
   }) => {
     const { publicKey, privateKey } = keyPair;
 
     const allAccounts = allAccountsSelector(sessionStore.getState());
     const accountName = `Account ${allAccounts.length + 1}`;
 
+    let activeHashKey = await getActiveHashKeyCryptoKey({ sessionStore });
+    if (activeHashKey === null && isSettingHashKey) {
+      // this should only happen on account creation & account recovery
+      activeHashKey = await deriveKeyFromString(password);
+    }
+
+    if (activeHashKey === null) {
+      throw new Error("Error deriving hash key");
```

Comment:
Short circuit if we unexpectedly don't have the hash key. The error will be picked up and reported by the caller